### ISSUE TITLE
Doesn't work in multi-folder workspaces

### ISSUE THREAD

**Issue by asvetliakov** [hashicorp/vscode-terraform#685]
vscode-terraform: 2.13.0
terraform-ls: 0.19.0

Scenario 1, have following VSCode workspace structure:
```json
{
  "folders": [
    {
      "name": "TF Project 1",
      "path": "tf1"
    },
    {
      "name": "TF Project 2",
      "path": "tf2"
    },
    {
      "name": "TF Project 3",
      "path": "tf3"
    }
  ]
}
```
*Result:*: doesn't work in anything but `TF Project 1`. I.e. it works only for the first folder, if i reorder and move Project 2 to the top, it's starting to work here, but not in Project 1

Scenario 2:
```json
{
  "folders": [
    {
      "name": "NON TF Project",
      "path": "nontf"
    },
    {
      "name": "TF Project 1",
      "path": "tf1"
    },
    {
      "name": "TF Project 2",
      "path": "tf2"
    },
    {
      "name": "TF Project 3",
      "path": "tf3"
    }
  ]
}
```

*Result:*": Doesn't work anywhere

**Comment by asvetliakov** [comment]
> what doesn't work
Nothing works. No completion, no formatting, no go-to-def, no hovers, etc

**Comment by radeksimko** [comment]
@asvetliakov Do you mind providing some more details? e.g. describe the hierarchy, attach configs, and/or any output from the relevant `Output` pane?

<img src="https://user-images.githubusercontent.com/287584/119831919-a3d76780-bef5-11eb-8ab1-15cdad58112b.png">

**Comment by asvetliakov** [comment]
Sure, here you go, complete repro: https://github.com/asvetliakov/tf-issue-685

1. clone
2. `terrform init` in proj2
3.  open workspace file
4. Navigate to `test.tf`
5. Try to add something to `aws_s3_bucket` or hover over `bucket` field - nothing works

Open just `proj2` instead of the workspace - everything is fine.

This is related to scenario 2, for scenario 1 just make a copy of proj2 and add it to the workspace.

**Comment by asvetliakov** [comment]
<details>
  <summary>And just in the case if you need the log from the output pane:</summary>
Launching language server: /home/alexey/.vscode-server/extensions/hashicorp.terraform-2.13.0/lsp/terraform-ls serve for folder: file:///home/alexey/work/tf-issue-685/proj1/
2021/07/13 22:46:51 serve_command.go:150: Starting terraform-ls 0.19.0
2021/07/13 22:46:51 service.go:82: Preparing new session ...
2021/07/13 22:46:52 watcher.go:246: watching for changes ...
2021/07/13 22:46:52 langserver.go:94: Starting server (pid 21023; concurrency: 8) ...
2021/07/13 22:46:52 server.go:591: Received 1 new requests
2021/07/13 22:46:52 server.go:175: Processing 1 requests
2021/07/13 22:46:52 server.go:266: Checking request for "initialize": {"processId":20913,"clientInfo":{"name":"vscode","version":"1.58.1"},"rootPath":"/home/alexey/work/tf-issue-685/proj1","rootUri":"file:///home/alexey/work/tf-issue-685/proj1","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"tagSupport":{"valueSet":[1]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true,"semanticTokens":{"refreshSupport":true}},"textDocument":{"publishDiagnostics":{"relatedInformation":true,"versionSupport":false,"tagSupport":{"valueSet":[1,2]},"codeDescriptionSupport":true,"dataSupport":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true,"tagSupport":{"valueSet":[1]},"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]}},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true},"activeParameterSupport":true},"contextSupport":true},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true,"tagSupport":{"valueSet":[1]},"labelSupport":true},"codeAction":{"dynamicRegistration":true,"isPreferredSupport":true,"disabledSupport":true,"dataSupport":true,"resolveSupport":{"properties":["edit"]},"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true,"prepareSupportDefaultBehavior":true},"documentLink":{"dynamicRegistration":true,"tooltipSupport":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true},"selectionRange":{"dynamicRegistration":true},"callHierarchy":{"dynamicRegistration":true},"semanticTokens":{"dynamicRegistration":true,"tokenTypes":["namespace","type","class","enum","interface","struct","typeParameter","parameter","variable","property","enumMember","event","function","member","macro","keyword","modifier","comment","string","number","regexp","operator"],"tokenModifiers":["declaration","definition","readonly","static","deprecated","abstract","async","modification","documentation","defaultLibrary"],"formats":["relative"],"requests":{"range":true,"full":{"delta":true}}}},"window":{"workDoneProgress":true}},"initializationOptions":{"commandPrefix":"g","experimentalFeatures":{}},"trace":"off","workspaceFolders":[{"uri":"file:///home/alexey/work/tf-issue-685/proj1","name":"proj1"},{"uri":"file:///home/alexey/work/tf-issue-685/proj2","name":"proj2"}]}
2021/07/13 22:46:52 rpc_logger.go:29: Incoming request for "initialize" (ID 0): {"processId":20913,"clientInfo":{"name":"vscode","version":"1.58.1"},"rootPath":"/home/alexey/work/tf-issue-685/proj1","rootUri":"file:///home/alexey/work/tf-issue-685/proj1","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true,"resourceOperations":["create","rename","delete"],"failureHandling":"textOnlyTransactional"},"didChangeConfiguration":{"dynamicRegistration":true},"didChangeWatchedFiles":{"dynamicRegistration":true},"symbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"tagSupport":{"valueSet":[1]}},"executeCommand":{"dynamicRegistration":true},"configuration":true,"workspaceFolders":true,"semanticTokens":{"refreshSupport":true}},"textDocument":{"publishDiagnostics":{"relatedInformation":true,"versionSupport":false,"tagSupport":{"valueSet":[1,2]},"codeDescriptionSupport":true,"dataSupport":true},"synchronization":{"dynamicRegistration":true,"willSave":true,"willSaveWaitUntil":true,"didSave":true},"completion":{"dynamicRegistration":true,"contextSupport":true,"completionItem":{"snippetSupport":true,"commitCharactersSupport":true,"documentationFormat":["markdown","plaintext"],"deprecatedSupport":true,"preselectSupport":true,"tagSupport":{"valueSet":[1]},"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]}},"completionItemKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}},"hover":{"dynamicRegistration":true,"contentFormat":["markdown","plaintext"]},"signatureHelp":{"dynamicRegistration":true,"signatureInformation":{"documentationFormat":["markdown","plaintext"],"parameterInformation":{"labelOffsetSupport":true},"activeParameterSupport":true},"contextSupport":true},"definition":{"dynamicRegistration":true,"linkSupport":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true,"symbolKind":{"valueSet":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26]},"hierarchicalDocumentSymbolSupport":true,"tagSupport":{"valueSet":[1]},"labelSupport":true},"codeAction":{"dynamicRegistration":true,"isPreferredSupport":true,"disabledSupport":true,"dataSupport":true,"resolveSupport":{"properties":["edit"]},"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"codeLens":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true,"prepareSupport":true,"prepareSupportDefaultBehavior":true},"documentLink":{"dynamicRegistration":true,"tooltipSupport":true},"typeDefinition":{"dynamicRegistration":true,"linkSupport":true},"implementation":{"dynamicRegistration":true,"linkSupport":true},"colorProvider":{"dynamicRegistration":true},"foldingRange":{"dynamicRegistration":true,"rangeLimit":5000,"lineFoldingOnly":true},"declaration":{"dynamicRegistration":true,"linkSupport":true},"selectionRange":{"dynamicRegistration":true},"callHierarchy":{"dynamicRegistration":true},"semanticTokens":{"dynamicRegistration":true,"tokenTypes":["namespace","type","class","enum","interface","struct","typeParameter","parameter","variable","property","enumMember","event","function","member","macro","keyword","modifier","comment","string","number","regexp","operator"],"tokenModifiers":["declaration","definition","readonly","static","deprecated","abstract","async","modification","documentation","defaultLibrary"],"formats":["relative"],"requests":{"range":true,"full":{"delta":true}}}},"window":{"workDoneProgress":true}},"initializationOptions":{"commandPrefix":"g","experimentalFeatures":{}},"trace":"off","workspaceFolders":[{"uri":"file:///home/alexey/work/tf-issue-685/proj1","name":"proj1"},{"uri":"file:///home/alexey/work/tf-issue-685/proj2","name":"proj2"}]}
2021/07/13 22:46:52 walker.go:180: asynchronously walking through /home/alexey/work/tf-issue-685/proj1
2021/07/13 22:46:52 walker.go:280: walking of /home/alexey/work/tf-issue-685/proj1 finished
2021/07/13 22:46:52 walker.go:186: async walking through /home/alexey/work/tf-issue-685/proj1 finished
2021/07/13 22:46:52 walker.go:180: asynchronously walking through /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 walker.go:228: found module /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 rpc_logger.go:50: Response to "initialize" (ID 0): {"capabilities":{"textDocumentSync":{"openClose":true,"change":2,"save":{}},"completionProvider":{"triggerCharacters":[".","["]},"hoverProvider":true,"signatureHelpProvider":{},"declarationProvider":{},"definitionProvider":true,"referencesProvider":true,"documentSymbolProvider":true,"codeLensProvider":{},"documentLinkProvider":{},"workspaceSymbolProvider":true,"documentFormattingProvider":true,"documentOnTypeFormattingProvider":{"firstTriggerCharacter":""},"executeCommandProvider":{"commands":["g.terraform-ls.module.callers","g.terraform-ls.rootmodules","g.terraform-ls.terraform.init","g.terraform-ls.terraform.validate"],"workDoneProgress":true},"semanticTokensProvider":{"legend":{"tokenTypes":["type","string","property","keyword","number","parameter","variable"],"tokenModifiers":["deprecated","modification"]},"full":true},"workspace":{"workspaceFolders":{"supported":true,"changeNotifications":"workspace/didChangeWorkspaceFolders"}}},"serverInfo":{"name":"terraform-ls","version":"0.19.0"}}
2021/07/13 22:46:52 server.go:247: Completed 1 requests [1.394ms elapsed]
2021/07/13 22:46:52 module_manager.go:70: MM: adding new module: /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 module_loader.go:214: ML: enqueing "OpTypeGetTerraformVersion" module operation: /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 module_loader.go:214: ML: enqueing "OpTypeObtainSchema" module operation: /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 walker.go:280: walking of /home/alexey/work/tf-issue-685/proj2 finished
2021/07/13 22:46:52 walker.go:186: async walking through /home/alexey/work/tf-issue-685/proj2 finished
2021/07/13 22:46:52 module_loader.go:148: executing "OpTypeObtainSchema" for /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 module_loader.go:148: executing "OpTypeGetTerraformVersion" for /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:52 server.go:591: Received 1 new requests
2021/07/13 22:46:52 server.go:175: Processing 1 requests
2021/07/13 22:46:52 server.go:266: Checking request for "initialized": {}
2021/07/13 22:46:52 rpc_logger.go:29: Incoming notification for "initialized": {}
2021/07/13 22:46:52 module_loader.go:159: finished "OpTypeGetTerraformVersion" for /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:46:53 provider_schema.go:106: PSS: adding local schema (/home/alexey/work/tf-issue-685/proj2, registry.terraform.io/hashicorp/aws): 0xc00614a018
2021/07/13 22:46:53 module_loader.go:165: finished "OpTypeObtainSchema" for /home/alexey/work/tf-issue-685/proj2
2021/07/13 22:47:26 server.go:591: Received 1 new requests
2021/07/13 22:47:26 server.go:175: Processing 1 requests
2021/07/13 22:47:26 server.go:266: Checking request for "workspace/executeCommand": {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:26 rpc_logger.go:29: Incoming request for "workspace/executeCommand" (ID 1): {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:26 rpc_logger.go:50: Response to "workspace/executeCommand" (ID 1): {"v":0,"callers":[]}
2021/07/13 22:47:26 server.go:247: Completed 1 requests [214.8µs elapsed]
2021/07/13 22:47:39 server.go:591: Received 1 new requests
2021/07/13 22:47:39 server.go:175: Processing 1 requests
2021/07/13 22:47:39 server.go:266: Checking request for "workspace/executeCommand": {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj1"]}
2021/07/13 22:47:39 rpc_logger.go:29: Incoming request for "workspace/executeCommand" (ID 2): {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj1"]}
2021/07/13 22:47:39 rpc_logger.go:50: Response to "workspace/executeCommand" (ID 2): {"v":0,"callers":[]}
2021/07/13 22:47:39 server.go:247: Completed 1 requests [109.6µs elapsed]
2021/07/13 22:47:48 server.go:591: Received 1 new requests
2021/07/13 22:47:48 server.go:175: Processing 1 requests
2021/07/13 22:47:48 server.go:266: Checking request for "workspace/executeCommand": {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:48 rpc_logger.go:29: Incoming request for "workspace/executeCommand" (ID 3): {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:48 rpc_logger.go:50: Response to "workspace/executeCommand" (ID 3): {"v":0,"callers":[]}
2021/07/13 22:47:48 server.go:247: Completed 1 requests [83.3µs elapsed]
2021/07/13 22:47:49 server.go:591: Received 1 new requests
2021/07/13 22:47:49 server.go:175: Processing 1 requests
2021/07/13 22:47:49 server.go:266: Checking request for "workspace/executeCommand": {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:49 rpc_logger.go:29: Incoming request for "workspace/executeCommand" (ID 4): {"command":"g.terraform-ls.module.callers","arguments":["uri=file:///home/alexey/work/tf-issue-685/proj2"]}
2021/07/13 22:47:49 rpc_logger.go:50: Response to "workspace/executeCommand" (ID 4): {"v":0,"callers":[]}
2021/07/13 22:47:49 server.go:247: Completed 1 requests [85.3µs elapsed]
</details>

**Comment by radeksimko** [comment]
This is helpful, thank you - I will try to reproduce and investigate.

**Comment by radeksimko** [comment]
I managed to reproduce it - currently looking for the root cause.

**Comment by radeksimko** [comment]
I can confirm that this affects workspaces where the first folder is _not_ parent folder of the other folders which contain Terraform code.

The main root cause is the document selector

https://github.com/hashicorp/vscode-terraform/blob/583bdf59a19d46a0f5e2325f0d7a11379e0bd337/src/extension.ts#L284-L285

I'm working on a fix.

**Comment by radeksimko** [comment]
This was fixed in #688 which I just released as part of `v2.13.1`. Can you try upgrading to that version and confirm it resolved the mentioned issues? Thanks.

**Comment by asvetliakov** [comment]
@radeksimko I confirm that `v2.13.1` fixes the issue. Thank you!
